gate-web/src/main/groovy/com/netflix/spinnaker/gate/security/SpinnakerUser.groovy

```diff
@@ -25,6 +25,6 @@ import java.lang.annotation.Target
 
 @Target([ElementType.PARAMETER])
 @Retention(RetentionPolicy.RUNTIME)
-@AuthenticationPrincipal(errorOnInvalidType = true)
+@AuthenticationPrincipal
```

Comment:
Since `/auth/user` is now unprotected, the first call from Deck is unauthenticated, and thus the `anonymousUser` gets stuffed in. `anonymousUser` cannot be cast to a Spinnaker `User`, and thus a HTTP 500 is thrown, instead of a HTTP 200 with a blank body.